tensorflow_quantum/python/optimizers/rotosolve_minimizer.py

```diff
@@ -0,0 +1,261 @@
+        # was found within tolerance.
+        'num_iterations',
+        # The number of iterations of the rotosolve update.
+        'num_objective_evaluations',
+        # The total number of objective
+        # evaluations performed.
+        'position',
+        # A tensor containing the last argument value found
+        # during the search. If the search converged, then
+        # this value is the argmin of the objective function.
+        # A tensor containing the value of the objective from
+        # previous iteration
+        'last_objective_value',
+        # Save the latest evalued value of the
+        # objective function
+        'objective_value',
+        # A tensor containing the value of the objective
+        # function at the `position`. If the search
+        # converged, then this is the (local) minimum of
+        # the objective function.
+        'tolerance',
+        # Define the stop criteria. Iteration will stop when the
+        # objective value difference between two iterations is
+        # smaller than tolerance
+        'solve_param_i',
+        # The parameter index where rotosolve is currently
+        # modifying. Reserved for internal use.
+    ])
+
+
+def _get_initial_state(initial_position, tolerance, expectation_value_function):
+    """Create RotosolveOptimizerResults with initial state of search."""
+    init_args = {
+        "converged": tf.Variable(False),
+        "num_iterations": tf.Variable(0),
+        "num_objective_evaluations": expectation_value_function.iter,
+        "position": tf.Variable(initial_position),
+        "objective_value": tf.Variable(0.),
+        "last_objective_value": tf.Variable(0.),
+        "tolerance": tolerance,
+        "solve_param_i": tf.Variable(0)
+    }
+    return RotosolveOptimizerResults(**init_args)
+
+
+def minimize(expectation_value_function,
+             initial_position,
+             tolerance=1e-8,
+             max_iterations=50,
+             name=None):
+    """Applies the rotosolve algorithm.
+
+    The rotosolve algorithm can be used to minimize a linear combination
+
+    of quantum measurement expectation values. See the following paper:
+
+    [arXiv:1903.12166](https://arxiv.org/abs/1903.12166), Ken M. Nakanishi.
+    [arXiv:1905.09692](https://arxiv.org/abs/1905.09692), Mateusz Ostaszewski.
+
+    ### Usage:
+
+    The following example demonstrates the Rotosolve optimizer attempting
+     to find the minimum for two qubit ansatz expectation value.
+
+    Here we show an example of optimize a function which consists summation of
+    a few sinusoids.
+
+    >>> n = 10  # Number of sinusoids
+    >>> coefficient = tf.random.uniform(shape=[n])
+    >>> min_value = -tf.sum(tf.abs(coefficient)) # The min value
+    >>> func = lambda x:tf.sum(tf.sin(x) * coefficient) # Define the function
+    >>> # Optimize the function with rotosolve, start with random parameters
+    >>> result =  rotosolve_minimizer.minimize(func, np.random.random(n))

```

Comment:
Great looking snippet, Much shorter. My only concern is that you don't show any output here. Most of our snippets show the expected output you get after running the snippet. Might want to add a line with

```
>>> result.last_objective_values
-0.123 whatever
```
This might also mean making coefficients a `tf.constant` with some fixed values instead of making it random.

Reply:
Fixed in 25ecc76a0a29c6dc9ee815d847c54478ebc5998c